builtin/logical/pki/crl_util.go

```diff
@@ -0,0 +1,179 @@
+package pki
+
+import (
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/vault/logical"
+)
+
+type revocationInfo struct {
+	CertificateBytes []byte `json:"certificate_bytes"`
+	RevocationTime   int64  `json:"revocation_time"`
+}
+
+var (
+	crlLifetime       = time.Hour * 72
+	revokeStorageLock = &sync.Mutex{}
```

Comment:
This should not be a global. If you have multiple PKI backends (e.g. prod-pki/ stage-pki/) you do not wan them to share this. It should probably be located in the struct of the backend.